code/modules/overmap/sectors.dm

```diff
@@ -96,9 +96,27 @@ var/global/list/known_overmap_sectors
 
 /obj/effect/overmap/visitable/proc/move_to_starting_location()
 	var/datum/overmap/overmap = global.overmaps_by_name[overmap_id]
-	start_x = start_x || rand(OVERMAP_EDGE, overmap.map_size_x - OVERMAP_EDGE)
-	start_y = start_y || rand(OVERMAP_EDGE, overmap.map_size_y - OVERMAP_EDGE)
-	forceMove(locate(start_x, start_y, overmap.assigned_z))
+	var/location
+
+	if(start_x || start_y)
+		location = locate(
+			start_x || rand(OVERMAP_EDGE, overmap.map_size_x - OVERMAP_EDGE),
+			start_y || rand(OVERMAP_EDGE, overmap.map_size_y - OVERMAP_EDGE),
```

Comment:
Doesn't make sense.

Reply:
Does. There is reason for it.
If one of them not set, use random one, otherwise it will spawn on null place or 0.

Reply:
If one of them is not set, use the logic below to select the other one, then do all the setting as previously. The way you broke it up both duplicates code and for some reason uses different selection logic depending on whether one or both is unset.

Reply:
I don't know how to make it like you suggest. This is how I wanted it to be designed:
`Custom start --> Random unchecked start  --|> Random checked start --> Random Unchecked start`

Reply:
A low-effort approach would be to turn this branch into `if(x && y)` and drop these `||`s, and that would be fine with me. Setting just one and not the other in earlier code makes no sense anyway, and seems fine to override.